src/Sarif.WorkItems/Extensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.CodeAnalysis.Sarif.WorkItems
+{
+    public static class Extensions
+    {
+        public static SarifWorkItemModel CreateWorkItemModel(
+            this SarifLog sarifLog, 
+            SarifWorkItemContext rootContext = null)
+        {
+            var sarifWorkItemModel = new SarifWorkItemModel();
+            sarifWorkItemModel.InitializeFromLog(sarifLog, rootContext);
```

Comment:
Thoughts on whether it would be better to have a ctor that takes these arguments, rather than a separate initialization method?

Reply:
I like the separate constructor better from a usability point, however, I can see circumstances where I may want to initialize data in this class without actually having the log, that data would then be used to fetch the log from some location, which could then be initialized.

Not really relevant for this extension method, which requires the log.  It feels like we might want an overload in the ctor as well?

---
In reply to: [381012807](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381012807) [](ancestors = 381012807)

Reply:
In your scenario you can defer creating the model until you have the log in hand. Unless I'm missing your point. (Also, in a separate comment, I changed my mind and suggested a static method `SarifWorkItemModel.CreateFromLog` rather than a ctor.)

---
In reply to: [381014605](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381014605) [](ancestors = 381014605,381012807)

Reply:
I've actually just put the init code in the ctor and deleted the now unnecessary Extensions class. This most accurately reflects our current needs. I typically use an initializer pattern in cases when I want to control error handling during init (rather than raising an exception in the ctor). Another issue that comes up is if you require a parameterless constructor for some reason (for example, because it is necessary to satisfy a new() constraint in a generic unsafe). Finally, initializers may be helpful if you want to support reuse in the object, rather than requiring a new instance be constructed. 

I made the simplifications that I did in the spirit of keeping the current code focused on immediate needs/implementation. Open to making adjustments as we drive more work/features into the pipeline.

---
In reply to: [381023477](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381023477) [](ancestors = 381023477,381014605,381012807)

Reply:
I have come to experience the same pattern in having to create an init() method such as connecting to KeyVault asynchronously since constructors cannot be async and avoid having to do a too much work in ctor.

Reply:
Good point.

---
In reply to: [382128203](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r382128203) [](ancestors = 382128203)